new Github Action: Update only modifiable data from GitHub

marianfoo opened the pull request:
As discussed in #9 I tried to create a solution.
I created an action and a new file '[update-gh.ts'](https://github.com/SAP-samples/artifact-of-the-month/blob/8ac966a70b75dbf51d42ca9f48d0bfbd7a5e4aca/.github/workflows/update-gh.yaml).

I have run it once in my fork:
https://github.com/marianfoo/artifact-of-the-month/commit/49f8a2f6fb285f47b38c22a4982dcd04ae0148b0

IObert commented on analyzer/src/update-gh.ts:
```diff
@@ -0,0 +1,86 @@
+    }
+
+    // get data from github
+    const artifactsGithub: Artifact[] = await Providers[2].get(currentTrendsJson)
```
This seems a bit dangerous to me as it would break if someone changes the order of the array elements. Please change it to a direct import of the GH provoider

marianfoo replied:
You are absolutely right here.
This is very risky and I should have changed it from the start.

IObert commented:
Thanks a lot for this PR 🚀 I added one comment, the rest looks good to me. 

But I do have one more ask: Can you change the branch of your Github Page (<https://marianfoo.github.io/artifact-of-the-month/#/new>) to "update-only-github" so that I can see this in action?

marianfoo commented:
Hi Marius,
Thanks for the feedback.
I'll try to set up an example again.
The action always included commits in the PR branch.
I then tested too much and hit the github rate limit.

marianfoo commented:
Hi @IObert ,
here is the current example https://marianfoo.github.io/artifact-of-the-month/?sap-language=en#/updated
Unfortunately with the new sorting, the ranks are bit confusing, right?
Is it may be better here to not show the ranks at all?
![image](https://user-images.githubusercontent.com/13335743/142865854-f8b9cfe2-401c-4a0d-9442-2a42aa1f9afc.png)

IObert commented:
Yes, I think this would be too confusing. I would rather prefer to update only the UI5 tags but not the rank. 

WDYT?

marianfoo commented:
Right.
The purpose of this PR is specifically not to update the ranks.
And the page "updated" should simply show the top 99 in descending order after the last update.

marianfoo commented:
Hi  @IObert 
still have a problem.
I can´t make it the Github Request work again.
Everytime i try to run it, there is a error that i´ve exceeded the rate limit.
I´ve waited more than a day but i still throws this error.
https://github.com/marianfoo/artifact-of-the-month/runs/4288494034?check_suite_focus=true
`(node:1608) UnhandledPromiseRejectionWarning: HttpError: You have exceeded a secondary rate limit. Please wait a few minutes before you try again.
    at /home/runner/work/artifact-of-the-month/artifact-of-the-month/node_modules/@octokit/request/dist-src/fetch-wrapper.js:68:27
    at processTicksAndRejections (internal/process/task_queues.js:95:5)`

Any ideas?

IObert commented:
Strange, sounds to me like a problem of GItHub in general.

marianfoo commented:
I might have tested too much.
Is it possible to add a wait time between requests to github?
https://docs.github.com/en/rest/overview/resources-in-the-rest-api#secondary-rate-limits
https://docs.github.com/en/rest/guides/best-practices-for-integrators#dealing-with-secondary-rate-limits

IObert commented:
Possibly, I'm not sure

marianfoo commented:
Hi @IObert 
i removed the rank from the updated page:
https://github.com/SAP-samples/artifact-of-the-month/pull/14/commits/2f189e4463ed91a241311257837afe09829e9bd8 https://github.com/SAP-samples/artifact-of-the-month/pull/14/commits/a856a7367245692ffefee17b8d868d4d31e535c9

implemented a rate limit throttle on octokit
https://github.com/SAP-samples/artifact-of-the-month/pull/14/commits/a6e8c8e04f7ef37582d35c2ea7eb448c30d15c30
I used this template
https://octokit.github.io/rest.js/v18#throttling

The new action ran succesfully 
Idid not run in rate limit issues, apperently the limit on [authenticated requests is 5.000 per hours](https://github.com/octokit/octokit.net/blob/main/Octokit/Exceptions/RateLimitExceededException.cs#:~:text=octokit%20%2F%20octokit.net%20Public&text=%2F%2F%2F%20unauthenticated%20requests%2C%20the%20rate,to%2060%20requests%20per%20hour.), which should be enough:
https://github.com/marianfoo/artifact-of-the-month/runs/4301373312?check_suite_focus=true

The changes are visible on my github pages site:
https://marianfoo.github.io/artifact-of-the-month/#/updated

IObert commented:
Awesome! Many thanks 🥳
